include/snitch/snitch_macros_test_case.hpp

```diff
@@ -36,24 +45,43 @@
-            {#FIXTURE, __VA_ARGS__}, SNITCH_CURRENT_LOCATION, []() { ID{}.test_fun(); });          \
-    void ID::test_fun()
-
-#define SNITCH_TEST_CASE_METHOD(FIXTURE, ...)                                                      \
-    SNITCH_TEST_CASE_METHOD_IMPL(                                                                  \
-        SNITCH_MACRO_CONCAT(test_fixture_, __COUNTER__), FIXTURE, __VA_ARGS__)
+#if !(SNITCH_DISABLE)
+#    define SNITCH_TEMPLATE_TEST_CASE(NAME, TAGS, ...)                                             \
+        SNITCH_TEMPLATE_TEST_CASE_IMPL(                                                            \
+            SNITCH_MACRO_CONCAT(test_fun_, __COUNTER__), NAME, TAGS, __VA_ARGS__)
+#else // SNITCH_DISABLE
+#    define SNITCH_TEMPLATE_TEST_CASE(NAME, TAGS, ...)                                             \
+        template<typename TestType>                                                                \
+        [[maybe_unused]] static void SNITCH_MACRO_CONCAT(test_id_, __COUNTER__)()
+#endif // SNITCH_DISABLE
+
+#if !(SNITCH_DISABLE)
+#    define SNITCH_TEST_CASE_METHOD_IMPL(ID, FIXTURE, ...)                                         \
+        namespace {                                                                                \
+        struct ID : FIXTURE {                                                                      \
+            void test_fun();                                                                       \
+        };                                                                                         \
+        }                                                                                          \
+        static const char* SNITCH_MACRO_CONCAT(test_id_, __COUNTER__) [[maybe_unused]] =           \
+            snitch::tests.add_fixture(                                                             \
+                {#FIXTURE, __VA_ARGS__}, SNITCH_CURRENT_LOCATION, []() { ID{}.test_fun(); });      \
+        void ID::test_fun()
+
+#    define SNITCH_TEST_CASE_METHOD(FIXTURE, ...)                                                  \
+        SNITCH_TEST_CASE_METHOD_IMPL(                                                              \
+            SNITCH_MACRO_CONCAT(test_fixture_, __COUNTER__), FIXTURE, __VA_ARGS__)
+#else // SNITCH_DISABLE
+#    define SNITCH_TEST_CASE_METHOD_IMPL(ID, FIXTURE, ...)                                         \
+        namespace {                                                                                \
+        struct ID : FIXTURE {                                                                      \
+            void test_fun();                                                                       \
+        };                                                                                         \
+        }                                                                                          \
+        inline void ID::test_fun()
```

Comment:
I don't think this `inline` is necessary?
```suggestion
        void ID::test_fun()
```